bpf/lbmap/ipv4.go

```diff
@@ -115,43 +147,53 @@ func Service4DumpParser(key []byte, value []byte) (bpf.MapKey, bpf.MapValue, err
 		return nil, nil, fmt.Errorf("Unable to convert key: %s\n", err)
 	}
 
-	svcKey.Port = common.Swab16(svcKey.Port)
-
 	if err := binary.Read(valueBuf, binary.LittleEndian, &svcVal); err != nil {
 		return nil, nil, fmt.Errorf("Unable to convert key: %s\n", err)
 	}
 
-	svcVal.Port = common.Swab16(svcVal.Port)
-	svcVal.RevNAT = common.Swab16(svcVal.RevNAT)
-
-	return &svcKey, &svcVal, nil
+	return svcKey.Convert(), svcVal.Convert(), nil
 }
 
-type RevNat4Key uint16
+type RevNat4Key struct {
+	Key uint16
+}
 
-func NewRevNat4Key(value uint16) RevNat4Key {
-	return RevNat4Key(common.Swab16(value))
+func NewRevNat4Key(value uint16) *RevNat4Key {
+	return &RevNat4Key{value}
 }
 
-func (k RevNat4Key) IsIPv6() bool           { return false }
-func (k RevNat4Key) Map() *bpf.Map          { return RevNat4Map }
-func (k RevNat4Key) NewValue() bpf.MapValue { return &RevNat4Value{} }
-func (k RevNat4Key) GetKeyPtr() unsafe.Pointer {
-	return unsafe.Pointer(&k)
+func (k *RevNat4Key) IsIPv6() bool              { return false }
+func (k *RevNat4Key) Map() *bpf.Map             { return RevNat4Map }
+func (k *RevNat4Key) NewValue() bpf.MapValue    { return &RevNat4Value{} }
+func (k *RevNat4Key) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(k) }
+func (k *RevNat4Key) String() string            { return fmt.Sprintf("%d", k.Key) }
+
+func (k *RevNat4Key) Convert() RevNatKey {
+	n := *k
+	n.Key = common.Swab16(n.Key)
+	return &n
 }
 
 type RevNat4Value struct {
 	Address types.IPv4
 	Port    uint16
 }
 
-func (k RevNat4Value) GetValuePtr() unsafe.Pointer {
-	return unsafe.Pointer(&k)
+func (v *RevNat4Value) GetValuePtr() unsafe.Pointer { return unsafe.Pointer(v) }
+
+func (v *RevNat4Value) Convert() RevNatValue {
+	n := *v
+	n.Port = common.Swab16(n.Port)
+	return &n
+}
+
+func (v *RevNat4Value) String() string {
```

Comment:
I've checked, remove the pointer, otherwise the printf doesn't work:
with pointer:
```go
rnv := RevNat4Value{}
fmt.Printf("%s\n", rnv) //Prints something like {%!s(int=2) %!s(int=3)}
```
Without pointer
```go
rnv := RevNat4Value{}
fmt.Printf("%s\n", rnv) //Prints something the returned value of the String() method
```